components/collections/src/codepointinvlist/cpinvlist.rs

```diff
@@ -74,17 +110,36 @@ impl databake::Bake for CodePointInversionList<'_> {
     }
 }
 
-// Note: serde(flatten) currently does not promote a struct field of type Vec
-// to replace the struct when serializing. The error message from the default
-// serialization is: "can only flatten structs and maps (got a sequence)".
-
 #[cfg(feature = "serde")]
 impl<'data> serde::Serialize for CodePointInversionList<'data> {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
-        self.inv_list.serialize(serializer)
+        if serializer.is_human_readable() {
+            use serde::ser::Error;
+            use serde::ser::SerializeSeq;
+            let mut seq = serializer.serialize_seq(Some(self.inv_list.len() / 2))?;
+            for range in self.iter_ranges() {
+                let start = char::from_u32(*range.start()).ok_or_else(|| {
+                    S::Error::custom(format!("Invalid code point {}", range.start()))
+                })?;
+                let end = char::from_u32(*range.end()).ok_or_else(|| {
+                    S::Error::custom(format!("Invalid code point {}", range.end()))
+                })?;
+                if start == end {
+                    seq.serialize_element(&start)?;
+                } else {
+                    seq.serialize_element(&format!("{start}-{end}"))?;
```

Comment:
Observation: if `-` is involved in the range (start or end or single), we could get strings like

- `"-"`
- `"--x"`
- `"x--"`

but this should be fine since we expect code points in positions 0 and 2, even if they are `-`

Reply:
yep